tc/core/polyhedral/memory_promotion.h

```diff
@@ -35,15 +35,25 @@ enum class AccessType : short { Read, Write };
 // ScheduleTree*), the user is responsible for maintaining the correspondance
 // between schedule tree positions and footprints.
 // Overapproximates one dimension by its lower bound, affine function of
-// parameters and schedule dimensions visible around the scope, and by a
-// constant size.
+// parameters and schedule dimensions visible around the scope, by a
+// constant size, and by a pair offset/stride for strided accesses.  If the
+// access is not strided, then "offset" is a zero expression and "stride" is 1.
+// The lowerBound and the size are computed after removing the eventual stride.
 struct ScopedFootprintDim {
  public:
-  ScopedFootprintDim(isl::aff lb, isl::val s) : lowerBound(lb), size(s) {}
+  ScopedFootprintDim(
+      isl::aff lowerBound,
+      isl::val size,
+      isl::aff offset,
+      isl::val stride)
+      : lowerBound(lowerBound), size(size), offset(offset), stride(stride) {}
 
  public:
   isl::aff lowerBound;
   isl::val size;
+
+  isl::aff offset;
+  isl::val stride;
```

Comment:
Why do you need the offsets and strides on each dimension individually?
Wouldn't it be better to store them together?

Reply:
Because we already have bounds and sizes for each dimension individually.  Storing some things individually and some other things collectively is confusing.  They are stored individually for cases when only _some_ dimensions can be approximated, which is useful when promoting indirect accesses.